### PR TITLE
Update Documentation for Object.prototype.toString

### LINKED ISSUE #3257: Issue with "Object.prototype.toString()": (short summary here please)
MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/toString

#### What did you expect to see?

Reference to, comparison with, and possibly recommendation for, String().

#### Did you test this? If so, how?

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/object/tostring`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/toString
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/object/tostring/index.html
* Last commit: https://github.com/mdn/content/commit/3b80812a1d4f68e57f35fca763f50bf3e5dfe997
* Document last modified: 2021-02-19T19:57:02.000Z

</details>

### PR DESCRIPTION
**Pull request by dev-nicolaos**
#### Summary
Updates the documentation for the `Object.prototype.toString` method to clarify some aspects and correct some statements. See PR comments for more details.

Shoutout to the folks at javascript.info for help groking this subject: https://javascript.info/object-toprimitive

#### Motivation
Ideally these changes will help readers better understand the circumstances in which this method is used, how it works, and its relationship to similar concepts/methods.

#### Supporting details
I'll add relevant links in individual PR comments as some are specific to certain lines changed.

#### Related issues
Hopefully the edits to the description section fix #3257

#### Metadata

This PR…

- [ ] Adds a new document
- [x] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/object/tostring/index.md [hunk 1]
```diff
@@ -51,45 +55,14 @@ o.toString(); // returns [object Object]
 
 ## Parameters
 
-For Numbers and BigInts `toString()` takes an optional parameter
```
Comment (dev-nicolaos) [inline files/en-us/web/javascript/reference/global_objects/object/tostring/index.md:54]: This part isn't relevant to this method, so I removed it, but left a note acknowledging that object's descended from `Object` may implement a different `toString` methods that do accept parameters.

File: files/en-us/web/javascript/reference/global_objects/object/tostring/index.md [hunk 2]
```diff
@@ -51,45 +55,14 @@ o.toString(); // returns [object Object]
...
-  numbers](https://en.wikipedia.org/wiki/Decimal),
-- 16 for [hexadecimal numbers](https://en.wikipedia.org/wiki/Hexadecimal).
+By default `toString()` takes no parameters. However, objects that inherit from `Object` may override it with their own implementation that do take parameters. For example, the `toString()` methods implemented by {{jsxref("Number")}} and {{jsxref("BigInt")}} take an optional `radix` parameter.
 
 ## Examples
 
 ### Overriding the default toString method
 
 You can create a function to be called in place of the default `toString()`
-method. The `toString()` method takes no arguments and should return a
```
Comment (dev-nicolaos) [inline files/en-us/web/javascript/reference/global_objects/object/tostring/index.md:90]: The default `Object.prototype.toString` method takes no arguments, but this section is about overriding it with a custom implementation, which can take arguments.

File: files/en-us/web/javascript/reference/global_objects/object/tostring/index.md [hunk 3]
```diff
@@ -51,45 +55,14 @@ o.toString(); // returns [object Object]
...
-string. The `toString()` method you create can be any value you want, but it
-will be most useful if it carries information about the object.
+method. The `toString()` function you create must return a primitive, otherwise it will be ignored.
```
Comment (dev-nicolaos) [inline files/en-us/web/javascript/reference/global_objects/object/tostring/index.md:65]: See step 5.B.ii from https://262.ecma-international.org/9.0/#sec-ordinarytoprimitive

**Review by Elchi3 — APPROVED** [review]
Sorry for being slow. This is a great change! Thanks a lot for your PR! 👍

**Comment by dev-nicolaos** [comment]
> Sorry for being slow. This is a great change! Thanks a lot for your PR! 👍

No worries, I saw you'd had a lot to work through. Thanks for the review!

**Pull request merged**